app/WebParsing/ArtSciParser.hs

```diff
@@ -85,7 +86,9 @@ getDeptList tags =
         extractDepartments tableTags =
             -- Each aTag consists of a start tag, text, and end tag
             let aTags = TS.partitions (tagOpenAttrNameLit "a" "href" (const True)) tableTags
-                depts = map (\t -> (TS.fromAttrib "href" $ head t, T.strip $ TS.innerText t)) aTags
+                depts = mapMaybe getDept aTags
+                getDept [] = Nothing
```

Comment:
Define this helper function in a `where` block; and make sure to provide a type signature.